src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -5113,6 +5156,22 @@ def _build_onedeploy_arm_url(params):
         )
     return params.cmd.cli_ctx.cloud.endpoints.resource_manager + base_url
 
+def _build_deploymentstatus_url(params, deployment_id):
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    client = web_client_factory(params.cmd.cli_ctx)
+    sub_id = get_subscription_id(params.cmd.cli_ctx)
+    if not params.slot:
+        base_url = (
+            f"subscriptions/{sub_id}/resourceGroups/{params.resource_group_name}/providers/Microsoft.Web/sites/"
+            f"{params.webapp_name}/deploymentStatus/{deployment_id}?api-version={client.DEFAULT_API_VERSION}"
+        )
+    else:
+        base_url = (
+            f"subscriptions/{sub_id}/resourceGroups/{params.resource_group_name}/providers/Microsoft.Web/sites/"
+            f"{params.webapp_name}/slots/{params.slot}/deploymentStatus/{deployment_id}"
```

Comment:
nit: can we just add the `slot` details to the url, instead of constructing the entire url twice?